File: JAVA/Geek and its Game of Coins.java
class Solution {
    public static int findWinner(int n, int x, int y) {
        boolean dp[] = new boolean[n + 1];
        if(n >= 1)
            dp[1] = true;
        for(int i = 2; i <= n; i++) {
            if(i - 1 >= 0 && dp[i - 1] == false)
                dp[i] = true;
            else if(i - x >= 0 && dp[i - x] == false)
                dp[i] = true;
            else if(i - y >= 0 && dp[i - y] == false)
                dp[i] = true;
        }
        return dp[n] == true ? 1 : 0;
    }
}

File: C++/Geek and its Game of Coins.cpp
class Solution {
  public:
    int wins(int n, int x, int y, vector<int>& dp){
        if(n == 0) return 0;
        if(dp[n] != -1) return dp[n];
        int ans = 0;
        int condition1 = (!(wins(n-1, x, y, dp)));
        int condition2 = 0, condition3 = 0;
        if(n-x >= 0) condition2 = (!(wins(n-x, x, y, dp)));
        if(n-y >= 0) condition3 = (!(wins(n-y, x, y, dp)));
        return dp[n] = condition1 | condition2 | condition3;
    }
    int findWinner(int n, int x, int y) {
        vector<int> dp(n+1, -1);
        return wins(n, x, y, dp);
     }
};
